File: Cores/DotNet/Optimization/RemoteFunctions/IntervalRemoteFunction.cs
using System.Collections.Generic;
using System.Linq;
using Newtonsoft.Json.Linq;

using OSOL.Extremum.Cores.DotNet.Arithmetics;

namespace OSOL.Extremum.Cores.DotNet.Optimization.RemoteFunctions
{
    public class IntervalRemoteFunction: RemoteFunction<Interval>
    {

        public IntervalRemoteFunction(string json, int port, string field) : base(json, port, field)
        {
            
        }

        public sealed override Interval Calculate(Dictionary<string, Interval> values)
        {
            string url = $"http://localhost:{this.Port}/process_request?field={this.Field}&scope=interval";
            url += values.Select(kvp => $"&{kvp.Key}={kvp.Value.ConvertToJson().ToString()}").Aggregate("", (p1, p2) => p1 + p2);
            var request = JObject.Parse(Client.DownloadString(url));

            return new Interval(request);
        }
    }

}

File: Cores/JVM/src/main/scala/OSOL/Extremum/Cores/JVM/Optimization/RemoteFunctions/IntervalRemoteFunction.scala
package OSOL.Extremum.Cores.JVM.Optimization.RemoteFunctions

import spray.json._
import OSOL.Extremum.Cores.JVM.Arithmetics.Interval
import com.github.kevinsawicki.http.HttpRequest

import scala.collection.JavaConverters

class IntervalRemoteFunction(override val json: String, override val port: java.lang.Integer, override val field: String)
  extends RemoteFunction[Interval](json, port, field) {

  final override def apply(values: Map[String, Interval]): Interval = {
    val params = Map("field" -> this.field, "scope" -> "interval") ++ values.map { case (k, v) => (k, v.toJson.toString) }
    val request = HttpRequest.get(s"http://localhost:${this.port}/process_request", JavaConverters.mapAsJavaMap(params), true)

    request.body().parseJson.convertTo[Interval]
  }

}